### PR TITLE
btcpay update to v1.3.3, NBXplorer to v2.2.16 + update option

### LINKED ISSUE #2682: BTCPayServer Update
https://blog.btcpayserver.org/btcpay-server-1-3-0/

### PR DESCRIPTION
**Pull request by openoms**
fix #2682

### CODE REVIEW

File: home.admin/config.scripts/bonus.btcpayserver.sh [hunk 1]
```diff
@@ -556,5 +560,73 @@ if [ "$1" = "0" ] || [ "$1" = "off" ]; then
...
+  REMOTE=$(git rev-parse "$UPSTREAM")
+
+  if [ $LOCAL = $REMOTE ]; then
+    TAG=$(git tag | sort -V | tail -1)
+    echo "# Up-to-date on version $TAG"
+  else
+    echo "# Pulling latest changes..."
+    sudo -u btcpay git pull -p
+    TAG=$(git tag | sort -V | tail -1)
+    echo "# Reset to the latest release tag: $TAG"
```
Comment (Kixunil) [inline home.admin/config.scripts/bonus.btcpayserver.sh:605]: There isn't `git verify-tag ...`. Is this an oversight or you generally don't verify things?

Comment (openoms) [reply home.admin/config.scripts/bonus.btcpayserver.sh:605]: git verification is implemented for c-lightning and joinmarket so far (+ where there are binaries used)

could add it here too just need to be sure that the same key is signing the releases

Comment (Kixunil) [reply home.admin/config.scripts/bonus.btcpayserver.sh:605]: I've been using same BTCPay key for a while and it worked well. Actually BTCPay is one of the easiest apps to upgrade.

Comment (openoms) [reply home.admin/config.scripts/bonus.btcpayserver.sh:605]: great, will look into it. Thanks for the suggestion!

Comment (openoms) [reply home.admin/config.scripts/bonus.btcpayserver.sh:605]: @Kixunil it looks like the commits and tags are signed with the personal key of @NicolasDorier. Do you have link for the pubkey you use or are you pulling it from a generic keyserver?

Comment (openoms) [reply home.admin/config.scripts/bonus.btcpayserver.sh:605]: ok nvm, got it. It is shared on keybase https://docs.btcpayserver.org/Vault/HowToVerify/#importing-nicolas-dorier-pgp-keys-only-first-time

Comment (openoms) [reply home.admin/config.scripts/bonus.btcpayserver.sh:605]: not sure why there is no signature found:
```
btcpay@raspberrypi:~/btcpayserver $ git reset --hard v1.3.1
HEAD is now at 6b96f9b4 bump
btcpay@raspberrypi:~/btcpayserver $ git verify-tag -v v1.3.1
object 6b96f9b446666e738e8245115ad935230ecf857c
type commit
tag v1.3.1
tagger nicolas.dorier <nicolas.dorier@gmail.com> 1635570032 +0900

1.3.1
error: no signature found
```

Comment (openoms) [reply home.admin/config.scripts/bonus.btcpayserver.sh:605]: when there is  a signature:
```
btcpay@raspberrypi:~/btcpayserver $ git log --show-signature
commit 6b96f9b446666e738e8245115ad935230ecf857c (HEAD -> master, tag: v1.3.1, origin/master, origin/HEAD)
gpg: Signature made Sat 30 Oct 2021 06:00:25 BST
gpg:                using RSA key AB4CFA9895ACA0DBE27F6B346618763EF09186FE
gpg: Good signature from "Nicolas Dorier <nicolas.dorier@gmail.com>" [unknown]
gpg: WARNING: This key is not certified with a trusted signature!
gpg:          There is no indication that the signature belongs to the owner.
Primary key fingerprint: AB4C FA98 95AC A0DB E27F  6B34 6618 763E F091 86FE
Author: nicolas.dorier <nicolas.dorier@gmail.com>
Date:   Sat Oct 30 14:00:25 2021 +0900

    bump
```

Comment (openoms) [reply home.admin/config.scripts/bonus.btcpayserver.sh:605]: can verify the commit, but don't know why I cannot just use the tag?

Comment (openoms) [reply home.admin/config.scripts/bonus.btcpayserver.sh:605]: ok, resolved with https://github.com/rootzoll/raspiblitz/pull/2683/commits/2df86783870a166f997b476e5ffeabd9fc8b813f

Comment (Kixunil) [reply home.admin/config.scripts/bonus.btcpayserver.sh:605]: Yes, I pulled it from keybase after cross-verifying from the sources keybase lists. I observed fingerprint AB4CFA9895ACA0DBE27F6B346618763EF09186FE

I use `git verify-commit "$(git rev-list -n 1 "$GIT_TAG")"`.

**Comment by openoms** [comment]
Thanks for the comments will continue to iterate.

**Comment by openoms** [comment]
Continued in #2708

**Pull request closed without merging**